File: app/src/main/java/leoneo7/bealawyer/base/Entry.java
package leoneo7.bealawyer.base;

/**
 * Created by ryouken on 2016/11/03.
 */

public class Entry {
    int id;
    String title;
    String image;
    String numbering;
    long date;
    String repeat;

    public Entry(int id, String title, String image, String numbering, long date, String repeat) {
        this.id = id;
        this.title = title;
        this.image = image;
        this.numbering = numbering;
        this.date = date;
        this.repeat = repeat;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public String getNumbering() {
        return numbering;
    }

    public void setNumbering(String numbering) {
        this.numbering = numbering;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public long getDate() {
        return date;
    }

    public void setDate(long date) {
        this.date = date;
    }

    public String getRepeat() {
        return repeat;
    }

    public void setRepeat(String repeat) {
        this.repeat = repeat;
    }
}

File: app/src/main/kotlin/leoneo7/bealawyer/base/Entry.kt
package leoneo7.bealawyer.base

import android.os.Parcel
import android.os.Parcelable

/**
 * Created by ryouken on 2016/11/03.
 */

data class Entry(var entryId: Int, var title: String, var image: String?, var numbering: String?,
                 var repeat: Int, var date: Long, var categoryId: Int, var categoryName: String) : Parcelable {

    constructor(source: Parcel) : this(source.readInt(), source.readString(), source.readString(), source.readString(), source.readInt(), source.readLong(), source.readInt(), source.readString())

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel?, flags: Int) {
        dest?.writeInt(entryId)
        dest?.writeString(title)
        dest?.writeString(image)
        dest?.writeString(numbering)
        dest?.writeInt(repeat)
        dest?.writeLong(date)
        dest?.writeInt(categoryId)
        dest?.writeString(categoryName)
    }

    companion object {
        @JvmField val CREATOR: Parcelable.Creator<Entry> = object : Parcelable.Creator<Entry> {
            override fun createFromParcel(source: Parcel): Entry = Entry(source)
            override fun newArray(size: Int): Array<Entry?> = arrayOfNulls(size)
        }
    }
}
